bowling-ball/Game.cs

```diff
@@ -2,16 +2,63 @@
 
 namespace BowlingBall
 {
+    
     public class Game
     {
-        public void Roll(int pins)
-        {
-            throw new NotImplementedException();
+        private int[] Rolls = new int[21];
+        private int currentRoll;
+        public void Roll(int pins)
+        {
+            Rolls[currentRoll++] = pins;
         }
 
-        public int GetScore()
+        public int GetScore()
+        {
+            int score = 0;
+            int roll = 0;
+            for (int frame = 1; frame <= 10; frame++)
+            {
+                if(IsStrike(roll))// strike bonus 2 next rolls so first check for strike.
+                {
+                    score += 10 + StrikeBonus(roll);
+                    roll++;
+                }
+                else if (IsSpare(roll))
+                {
+                    score += 10 + SpareBonus(roll);
+                    roll += 2;

```

Comment:
It is always recommended to add a comment in case you are adding any logic specific to a number. This will help to understand logic even if no one is aware of the problem statement.